SU2_CFD/src/iteration_structure.cpp

```diff
@@ -107,8 +110,36 @@ void CIteration::SetGrid_Movement(CGeometry ****geometry_container,
 
       break;
 
-    case DEFORMING:
 
+    case ELASTICITY:
+
+      if (ExtIter != 0) {
+
+        if (rank == MASTER_NODE)
+          cout << " Deforming the grid using the Linear Elasticity solution." << endl;
+
+        /*--- Update the coordinates of the grid using the linear elasticity solution. ---*/
+        for (iPoint = 0; iPoint < geometry_container[val_iZone][val_iInst][MESH_0]->GetnPoint(); iPoint++) {
+
+          su2double *U_time_nM1 = solver_container[val_iZone][val_iInst][MESH_0][FEA_SOL]->node[iPoint]->GetSolution_time_n1();
+          su2double *U_time_n   = solver_container[val_iZone][val_iInst][MESH_0][FEA_SOL]->node[iPoint]->GetSolution_time_n();
+
+          for (iDim = 0; iDim < geometry_container[val_iZone][val_iInst][MESH_0]->GetnDim(); iDim++)
+            geometry_container[val_iZone][val_iInst][MESH_0]->node[iPoint]->AddCoord(iDim, U_time_n[iDim] - U_time_nM1[iDim]);
+
+        }
+
+      }
+
+      break;
+
+ 	/*--- Already initialized in the static mesh movement routine at driver level. ---*/ 
+    case STEADY_TRANSLATION: case ROTATING_FRAME:
+      break;
+
+  }
+  
+  if (config_container[val_iZone]->GetSurface_Movement(DEFORMING)){
```

Comment:
One thing I have been meaning to change is to remove the calls to SetVolume_Deformation() in between each of the updates of the surface positions.. in the case that multiple sequential surface movements are present, they should be additive (they use an increment to the position) and we should be able to call SetVolume_Deformation() just once at the end of the sequence. Can you please give it a look and adjust if you agree?

Reply:
Maybe we can deal with that in the new mesh solver implementation @rsanfer ?

Reply:
Agreed. Let's wait on this change. I will make an issue instead.